packages/rest/test/unit/deserializer.test.ts

```diff
@@ -0,0 +1,193 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {deserialize} from '../..';
+import {expect, ShotRequest, ShotRequestOptions} from '@loopback/testlab';
+import {OperationObject, ParameterObject} from '@loopback/openapi-spec';
+
+// tslint:disable:no-any
+describe('deserializer', () => {
+  function expectToDeserialize(
+    param: ParameterObject,
+    source: any[],
+    target: any[],
+  ) {
+    expect(source.map(i => deserialize(i, param))).to.eql(target);
+  }
+
+  function expectToDeserializeNullOrUndefined(param: ParameterObject) {
+    expect(deserialize(null, param)).to.be.null();
+    expect(deserialize(undefined, param)).to.be.undefined();
+  }
+
+  function expectToFail(
+    param: ParameterObject,
+    source: any[],
+    reason: string | RegExp,
+  ) {
+    for (const i of source) {
+      expect(() => deserialize(i, param)).to.throw(reason);
+    }
+  }
+
+  it('converts number parameters', () => {
+    const param: ParameterObject = {
+      in: 'query',
+      name: 'balance',
+      type: 'number',
+    };
+    expectToDeserialize(param, [0, 1.5, '10', '2.5'], [0, 1.5, 10, 2.5]);
+    expectToDeserializeNullOrUndefined(param);
+    expectToFail(
```

Comment:
One logical assert per test please. This should be split into several tests grouped by `describe` block.

```ts
describe('number parameter', () => {
 const param: ParameterObject = {
     in: 'query',
     name: 'balance',
     type: 'number',
   };

  it('deserializes integer values', () => {
    expectToDeserialize(param, [0, 10, -10], [0, 10, -10]);
  });

  it('deserializes integer strings', () => {
    expectToDeserialize(param, ['0', '10', '-10'], ['0', '10', '-10']);
  });

  // etc.
});
```

The same comment applies to other new tests added below.